app/src/main/java/com/github/se/stepquest/map/LocationArea.kt

```diff
@@ -32,78 +37,90 @@ class LocationArea {
+                  }
 
-                // Retrieve the list of Checkpoints
-                val checkpointsSnapshot = routeID.child("checkpoints")
-                val checkpointsList = mutableListOf<Checkpoint>()
-                for (checkpointSnapshot in checkpointsSnapshot.children) {
-                  val name = checkpointSnapshot.child("name").getValue<String>().orEmpty()
-                  val locationSnapshot = checkpointSnapshot.child("location")
-                  val latitude = locationSnapshot.child("latitude").getValue<Double>()
-                  val longitude = locationSnapshot.child("longitude").getValue<Double>()
-                  if (latitude != null && longitude != null) {
-                    val locationDetails = LocationDetails(latitude, longitude)
-                    checkpointsList.add(Checkpoint(name, locationDetails))
+                  // Retrieve the list of Checkpoints
+                  val checkpointsSnapshot = routeID.child("checkpoints")
+                  val checkpointsList = mutableListOf<Checkpoint>()
+                  for (checkpointSnapshot in checkpointsSnapshot.children) {
+                    val name = checkpointSnapshot.child("name").getValue<String>().orEmpty()
+                    val locationSnapshot = checkpointSnapshot.child("location")
+                    val latitude = locationSnapshot.child("latitude").getValue<Double>()
+                    val longitude = locationSnapshot.child("longitude").getValue<Double>()
+                    if (latitude != null && longitude != null) {
+                      val locationDetails = LocationDetails(latitude, longitude)
+                      checkpointsList.add(Checkpoint(name, locationDetails))
+                    }
                   }
-                }
 
-                // Retrieve the userID
-                val userID = routeID.child("userid").getValue<String>().orEmpty()
+                  // Retrieve the userID
+                  val userID = routeID.child("userid").getValue<String>().orEmpty()
 
-                // Create RouteDetails object
-                val routeDetailData =
-                    RouteDetails(routeKey, routeDetailsList, checkpointsList, userID)
+                  // Create RouteDetails object
+                  val routeDetailData =
+                      RouteDetails(routeKey, routeDetailsList, checkpointsList, userID)
 
-                // Add to routeDetailList
-                routeDetailList.add(routeDetailData)
+                  // Add to routeDetailList
+                  routeDetailList.add(routeDetailData)
+                }
+                insideArea = false
               }
-              insideArea = false
+              Log.i("Location Area", "RouteList: $routeList")
+              Log.i("Location Area", "RouteDetailList: $routeDetailList")
+
+              cacheRouteData(context, routeList, routeDetailList)
+
+              //            getcacheRouteData(context).let { (routeList, routeDetailList) ->
```

Comment:
There is still some commented code here it might be time to look into it and see if this code is still necessary or we can remove it

Reply:
yes we can delete that, thanks for noticing